tools/js-optimizer.js

```diff
@@ -8152,6 +8152,17 @@ function emitDCEGraph(ast) {
   var exportNameToGraphName = {}; // identical to asm['..'] nameToGraphName
   var foundAsmLibraryArgAssign = false;
   var graph = [];
+
```

Comment:
please update the js-optimizer test in other to test these new patterns directly.

Reply:
Updated with a test (although I think this is an instance where the test will add redundant cycles, as this is already tested in the functional tests for minimal runtime in test_core. Perhaps it will help in debugging however)